crates/wasi/Cargo.toml

```diff
@@ -12,9 +12,9 @@ keywords.workspace = true
 categories.workspace = true
 
 [dependencies]
-wasi-common = "2.0"
-wasi-cap-std-sync = "2.0"
-wiggle = { version = "2.0", default-features = false, features = ["wiggle_metadata"] }
+wasi-common = "17.0"
```

Comment:
Looking at `Cargo.toml` of the `wasi-common` crate's most recent version it has the `wasmtime` crate feature enabled as default feature. In order to avoid building Wasmtime via Wasmi we must disable this crate feature by using `default-features = false` and `features = ["wiggle_metadata", "trace_log", "sync"]`.